src/dbsync/src/dbsync_implementation.cpp

```diff
@@ -55,6 +55,24 @@ void DBSyncImplementation::syncRowData(const DBSYNC_HANDLE  handle,
                                       callback);
 }
 
+void DBSyncImplementation::syncRowData(const DBSYNC_HANDLE  handle,
+                                       const TXN_HANDLE     txn,
+                                       const char*          jsonRaw,
+                                       const ResultCallback callback)
+{
+    const auto& ctx{ dbEngineContext(handle) };
+    const auto& tnxCtx { ctx->transactionContext(txn) };
+    const auto json { nlohmann::json::parse(jsonRaw) };
+    if (std::find(tnxCtx->m_tables.begin(), tnxCtx->m_tables.end(), json[0]["table"]) == tnxCtx->m_tables.end())
+    {
+        throw dbsync_error{INVALID_TABLE};
+    }
+    ctx->m_dbEngine->syncTableRowData(json[0]["table"],
```

Comment:
Change the access from [] to at(), to prevent sig abort when the "table" or "data" does not exist in the input JSON, 

at() access generate a controlled exception.

make the change in lines 44, 53, 66, and add unit tests to cover this!